meveo-admin/ejbs/src/main/java/org/meveo/service/base/BusinessService.java

```diff
@@ -169,6 +179,87 @@ public BusinessEntity findByEntityClassAndCode(Class<?> clazz, String code) {
         }
     }
 
+    /** 
+     * Find which the module the entity belongs to 
+
+     * @param entity
+     * @return MeveoModule, or null otherwise
+     */
+    @SuppressWarnings({ "rawtypes", "deprecation" })
+	public MeveoModule findModuleOf(P entity){
+    	MeveoModule module = null;
+    	if (entity != null) {
+    		Session session = this.getEntityManager().unwrap(Session.class);
+    		Query q = session.createQuery("SELECT m.module FROM ModuleItem m WHERE m.code = :code AND m.ItemType = :itemType");
+    		q.setString("code", entity.getCode());
+    		q.setString("itemType", entity.getClass().getName());
+    		module = (MeveoModule) q.getResultList().get(0);
+    	}
+    	return module;
+    }
+    
+    /**
+     * Remove the entity that belongs to the module
+     * 
+     * @param entity
+     * @param module
+     */
+    public void removeFilesFromModule(P entity, MeveoModule module) {
+    	MeveoModuleDto meveoModuleDto = new MeveoModuleDto(module);
+    	BusinessEntityDto businessEntityDto = new BusinessEntityDto(entity);
+    	File fileToDelete = new File("..\\git\\"+meveoModuleDto.getCode()+"\\"+businessEntityDto.getClass().getAnnotation(ModuleItem.class).path()+"\\"+entity.getCode()+".json");
+
+    	fileToDelete.delete();
+    }
+    
+    /**
+     * Create the entity in the dedicated module
+     * 
+     * @param entity
+     * @param module
+     */
+    public void addFilesToModule(P entity, MeveoModule module) {
+    	MeveoModuleDto meveoModuleDto = new MeveoModuleDto(module);
+    	BusinessEntityDto businessEntityDto = new BusinessEntityDto(entity);
+    	String businessEntityDtoSerialize = JacksonUtil.toString(businessEntityDto);
+    	
+    	try {
+	    	File file = new File("..\\git\\"+meveoModuleDto.getCode()+"\\"+businessEntityDto.getClass().getAnnotation(ModuleItem.class).path()+"\\"+entity.getCode()+".json");
+	    	file.createNewFile();
+    	}catch(IOException e) {
+    		e.printStackTrace();
+    	}
+    	
+    	try {
+    		Path path = Paths.get("..\\git\\"+meveoModuleDto.getCode()+"\\"+businessEntityDto.getClass().getAnnotation(ModuleItem.class).path()+"\\"+entity.getCode()+".json");
+    		byte[] strToBytes = businessEntityDtoSerialize.getBytes();
+
+    		Files.write(path, strToBytes);
+    	}catch(IOException e) {
+    		e.printStackTrace();
+    	}
+    }
+    
+    /**
+     * Retrieve the current module of the entity, ignore if none
+     * Remove the files from old Module
+     * Add the files to the new module
+     * 
+     * @param entity
+     * @param module
+     */
+    public void moveFilesToModule(P entity, MeveoModule module) {
+    	MeveoModule currentModule = findModuleOf(entity);
+    	if (currentModule != null) {
+    		removeFilesFromModule(entity, currentModule);
+    		
+	    	if ((module != null)) {
```

Comment:
You shouldn't make the null check here : if the module variable is null, it must fail